src/hashmap_variant.rs

```diff
@@ -0,0 +1,60 @@
+use std::{collections::HashMap, hash::Hash, mem};
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum SecondLayerMap<K, V>
```

Comment:
It would be nice to have a doc comment explaining for the reader what this is, and why it's there.

Reply:
I went ahead and added docs for that whole file. Let me know if everything looks good to you